prover/strategies/matching.md

```diff
@@ -177,6 +195,279 @@ Recurse over assoc-only constructors (including `pto`):
     requires T =/=K P
      andBool P in Vs
      andBool getReturnSort(T) ==K getReturnSort(P)
+
+
+  // A lot of repetetive code below.
+  // This could be reduced if we had a generic support
+  // for notations.
+
+  // \equals(_,_) matched
+  rule #matchAssoc( terms:     \equals(T1, T2), Ts
+                               => T1 ++Patterns T2 ++Patterns Ts
+                  , pattern:   \equals(P1, P2), Ps
+                               => P1 ++Patterns P2 ++Patterns Ps
+                  , variables: _
+                  , subst:     _
+                  , rest:      _
+                  )
+
+  // \equals(_,_) mismatched
+  rule #matchAssoc( terms:     T, _
+                  , pattern:   \equals(...), _
+                  , variables: _
+                  , subst:     _
+                  , rest:      _
+                  )
+    => #matchFailure("\\equals(_,_) does not match"), .MatchResults
+    requires \equals(...) :/=K T
+
+  // \not(_) matched
+  rule #matchAssoc( terms:     \not(T), Ts
+                               => T ++Patterns Ts
+                  , pattern:   \not(P), Ps
+                               => P ++Patterns Ps
+                  , variables: _
+                  , subst:     _
+                  , rest:      _
+                  )
+
+  // \not(_) mismatched
+  rule #matchAssoc( terms:     T, _
+                  , pattern:   \not(_), _
+                  , variables: _
+                  , subst:     _
+                  , rest:      _
+                  )
+    => #matchFailure("\\not(_) does not match"), .MatchResults
+    requires \not(...) :/=K T
```

Comment:
I'm not sure if this is sound

Reply:
Do you mean this general pattern that is used also for \equals and others, or do you mean this particular instance? Why it is not sound?

Reply:
Do you mean the general pattern that is used also for \equals and others, or do you mean this particular instance? Why it is not sound?

Reply:
So when we implemented this, we only considered constructors possibly modulo AC. I'm not sure if generalizing #match so much makes sense.

Reply:
Perhaps we want to simplify the pattern to remove instances of and, or, not etc before called #match?

Reply:
So maybe having a separate version `#syntacticalMatch` would resolve the issue?